### PR TITLE
feat: add Qwen3.5 support

### PR DESCRIPTION
**Pull request by JohannesHa**
Bump vLLM to >=0.16.1.dev (nightly) which includes Qwen3.5 model support. This requires torch 2.10 (resolved from the existing >=2.9.0 pin), an updated flash-attn wheel built against torch 2.10, and version overrides for nvidia-cutlass-dsl and quack-kernels.

Bump transformers pin to 5c1c72b which includes a rope validation fix for Qwen3.5 (https://github.com/huggingface/transformers/pull/44272).

Add a trainer monkey-patch for a transformers bug where Qwen3.5 passes 3D MRoPE position_ids to decoder layers instead of 2D text_position_ids, which breaks flash attention and causes NaN gradients. The upstream fix is pending: https://github.com/huggingface/transformers/pull/44399

---

> [!NOTE]
> **Medium Risk**
> Medium risk due to major dependency upgrades (torch/vLLM/transformers) and runtime monkey-patching of model internals, which could affect training stability or inference behavior across supported models.
> 
> **Overview**
> Enables Qwen3.5 support by upgrading the runtime stack: `vllm` moves to a nightly `0.16.1` dev build (new `vllm-nightly` index), `torch`/`torchaudio`/`torchvision` bump to the CUDA 12.8 stable channel, `transformers` advances to a newer pinned git rev, and the `flash-attn` wheel is updated for torch 2.10; `uv` overrides are expanded (notably `quack-kernels`) and the lockfile is refreshed accordingly.
> 
> Updates the token-based chat serving path (`serving_chat_with_tokens.py`) to match newer vLLM APIs (no explicit logits-processor validation / no manual `engine_request` construction), while still forcing `prompt_token_ids` and correctly computing `reasoning_ended` from the provided token IDs.
> 
> Adds Qwen3.5-specific trainer monkey patches in `trainer/model.py` to fix upstream `transformers` issues (3D MRoPE `position_ids` handling and incorrect Qwen3.5 MoE checkpoint conversion mapping), and extends VLM detection plus perf accounting to handle Qwen3.5 and configs that use `moe_intermediate_size`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 1cdea1b3609de32bff6f9b0c7106c55db3a0372d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/prime_rl/utils/vlm.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 # Add new patterns here as they are tested and supported
 SUPPORTED_VLM_PATTERNS = [
     "Qwen/Qwen3-VL*",
+    "Qwen/Qwen3.5*",
```
[the remark this answers is not in the document]
Comment (JohannesHa) [reply src/prime_rl/utils/vlm.py:12]: there are no text only qwen3.5s afaik

**Review by mikasenghaas — APPROVED** [review]
